eventsdiscovery/src/main/java/com/schedjoules/client/eventsdiscovery/http/SingleEventResponseHandler.java

```diff
@@ -56,21 +58,14 @@ public String etag()
             @Override
             public String uid()
             {
-                return payload().uid();
+                return payload().value().uid();
```

Comment:
I don't think so. Both envelopes are generated differently. `MultiEventsResponseHander` takes the ETAG from the JSON response, whereas `SingleEventResponseHandler` takes the ETAG from a header.
Also in a multiresponse events are actually optional (they may be absent if the event was removed), but a single event response the event is not optional. If the event doesn't exist the request will fail with a `NotFoundException`.